File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        nums=set(nums)
        longest = 0

        for n in nums:
            if (n - 1) not in nums:
                length = 1
                while (n + length) in nums:
                    length += 1
                longest = max(length, longest)
        return longest

File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        unordered_set<int> s(nums.begin(), nums.end());
        int longest = 0;

        for(auto it: s){
            if(s.find(it-1) == s.end()){
                int length = 1;
                while (s.find(it+length) != s.end()){
                    length++;
                }
                longest = max(longest, length);
            }
        }

        return longest;
    }
};